webpack/ForemanTasks/Components/Chart/Chart.js

```diff
@@ -38,7 +38,12 @@ class C3Chart extends React.Component {
 
   destroyChart() {
     try {
-      this.chart = this.chart.destroy();
+      // A workaround for a case, where the chart might be still in transition
+      // phase while unmounting/destroying - destroying right away leads
+      // to issue described in https://github.com/bcbcarl/react-c3js/issues/22.
+      // Delaying the destroy a bit seems to resolve the issue.
+      setTimeout(this.chart.destroy, 1000);
+      this.chart = null;
```

Comment:
When you pass functions this way in javascript, you loose the `this` reference. So if the `destroy` function would use the `this` keywork it will receive something different than the chart.

Therefore I want to suggest using:
```js
setTimeout(() => {
  this.chart.destroy();
  this.chart = null;
}, 1000);
```

Reply:
Notice that we're actually not adding the `this.chart = null` within timeout, which is on purpose (so that we make sure this component is not touching it after that point). I don't think we care about `this` binding within the timeout

Reply:
I don't think `this.chart = null` will actually mean something because the component itself is already destroyed.

The `this` binding work because of the implementation of `chart.destroy` don't use `this`. Once they will this code will break.

Reply:
I see. Actually, the `destroy` function is using `this` already and it works thanks to the fact that `c3` is binding the API functions explicitly https://github.com/c3js/c3/blob/062fc2a93915b94b36daa1e168334109f5edcf20/src/chart.js#L17-L24

On the other hand, I don't know how the component behaves after it's unmounted (and would be referenced later in after timeout). Therefore, I would probably just keep this is it is.

Reply:
I've added an explicit comment though the mention that it's ok to do it this way in this case